### PR TITLE
gnucash: new recipe

### PR DESCRIPTION
**Pull request by jsteinaker**
Clearly a work in progress, as we are still trying to get the dependencies properly packaged. It's missing the application icon, Deskbar entry, etc.

Guile crashes during the build process, I have to run it a few times until at some point it compiles. Very possibly related to the crash I'm getting when opening Gnucash (which still leaves the application open and working). Debugger points to libgc, I'm suspicious of Guile thread support. I'm trying to build Guile without threads but I'm failing miserably so far.

### CODE REVIEW

File: app-office/gnucash/gnucash-5.9.recipe [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	lib:libdbdsqlite3$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+    cmd:cmake
+    cmd:gcc$secondaryArchSuffix
+    cmd:gettext
+    cmd:git
+    cmd:ninja
+    cmd:pkg_config$secondaryArchSuffix
+    cmd:swig
```
Comment (jmairboeck) [inline app-office/gnucash/gnucash-5.9.recipe:119]: Use tabs for indentation.

File: app-office/gnucash/gnucash-5.9.recipe [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	devel:libsqlite3$secondaryArchSuffix
+	devel:libwebkit2gtk_4.1$secondaryArchSuffix
+	devel:libxml2$secondaryArchSuffix
+	devel:libxslt$secondaryArchSuffix
+	lib:libdbdsqlite3$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+    cmd:cmake
+    cmd:gcc$secondaryArchSuffix
+    cmd:gettext
```
Comment (jmairboeck) [inline app-office/gnucash/gnucash-5.9.recipe:115]:
```suggestion
    cmd:gettext$secondaryArchSuffix
```

File: app-office/gnucash/gnucash-5.9.recipe [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	gnucash = $portVersion
```
Comment (jmairboeck) [inline app-office/gnucash/gnucash-5.9.recipe:26]:
```suggestion
	gnucash$secondaryArchSuffix = $portVersion
```

File: app-office/gnucash/gnucash-5.9.recipe [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+	lib:libsqlite3$secondaryArchSuffix
+	lib:libwebkit2gtk_4.1$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
+	lib:libxslt$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	cmd:xsltproc$commandSuffix
```
Comment (jmairboeck) [inline app-office/gnucash/gnucash-5.9.recipe:87]: I would move this to `BUILD_PREREQUIRES`.

File: app-office/gnucash/gnucash-5.9.recipe [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    cmd:swig
+	"
+
+BUILD()
+{
+	cmake -B build -S . -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+		-DCMAKE_INSTALL_PREFIX=$prefix \
+		-DCMAKE_INSTALL_DATAROOTDIR=$dataDir \
+		-DCMAKE_INSTALL_INCLUDEDIR=$includeDir \
+		-DCMAKE_INSTALL_SYSCONFDIR=$prefix/settings \
```
Comment (jmairboeck) [inline app-office/gnucash/gnucash-5.9.recipe:128]:
```suggestion
		-DCMAKE_INSTALL_SYSCONFDIR=$settingsDir \
```

**Comment by jsteinaker** [comment]
Thanks for your help with all these recipes @jmairboeck

**Comment by jsteinaker** [comment]
I've added the icon an app signature. Not the best job keeping the icon small but it works. Not too sure about the app signature, I just did what made sense to me.

File: app-office/gnucash/gnucash-5.9.recipe [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	gnucash$secondaryArchSuffix = $portVersion
+	cmd:finance_quote_wrapper$commandSuffix
+	cmd:gnc_fq_update$commandSuffix
+	cmd:gnucash$commandSuffix
```
Comment (jmairboeck) [inline app-office/gnucash/gnucash-5.9.recipe:32]: This is moved to `$appsDir` below, so

```suggestion
	app:GnuCash
```
and sort it before the other `cmd`s.

**Comment by waddlesplash** [comment]
Do you have a crash report handy? Unless there's platform-specific code that's broken, it could be a Haiku bug somehow.

**Comment by jsteinaker** [comment]
I'll try to generate one for you soon, but for starters, I strongly believe that the crash is related to #11325, which I suspect is indeed a bug in Haiku, since downgrading Boehm-GC to 8.2.6 doesn't help and Guile was building just fine against it not long ago.

**Comment by jsteinaker** [comment]
Crash report

**Comment by waddlesplash** [comment]
Appears there's also https://github.com/haikuports/haikuports/issues/3851 which may be related.

**Comment by jsteinaker** [comment]
Using -DCMAKE_BUILD_TYPE=Release breaks the build because of -O3. I went with MinSizeRel, which as a bonus generates a package less than half the size (that you can probably achieve with RelWithDebInfo and stripping afterwards, to be honest). If someone with knowlegde about C compilers thinks there's a better way to manage this, please let me know.

**Comment by Begasus** [comment]
@jsteinaker what's the status on this one? Haven't checked myself in a while.

**Comment by jsteinaker** [comment]
It should be ready to merge. It's no longer crashing after the fixes in Boehm-GC. Unless someone has a good reason not to use MinSizeRel in CMake (which means -Os instead of -O3) I'd say it's good to go.

**Pull request merged**